### PR TITLE
Blaze: Show the promote with blaze card

### PR DESCRIPTION
**Pull request by zwarm**
Parent #17909 

This PR implements the Promote with Blaze entry point as a dashboard card

- Moves the PromoteWithBlazeCard to a DashboardCard so that it can be properly slotted within that group
- Moves location of ViewHolder 
- Creates the PromoteWithBlazeUpdate partial state
- Creates the builder params and the builder for the card
- Creates the PromoteWithBlazeSource and adds it the MySiteSourceManager
- Connects to the BlazeStore to get blaze status for the site
- Updates and adds new unit tests
- If the dashboard cards are in error, then the blaze card won't be shown as well

**Notes**
- There are `todo` comments in `MySiteViewModel` for implementing the clicks functionality and the tracking events. This PR was getting too big, so these tasks will be handled in a separate PR.
- Launching the blaze overlay is blocked until the overlay code is ready and is in trunk
- The **UI needs to be tidied up**, this will follow in a separate PR

Light | Dark
-- | --
Alt desc | Alt desc

**Merge Instructions**
This PR is dependent upon [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2659)
- Wait for the FluxC PR to be successfully merged into trunk
- Copy the trunk hash (from the PR) and apply it to build.gradle in this PR
- Push the commit
- Wait for CI to complete
- Merge as normal
**OR** - Approve the PRs only and I will merge :)

**Test**
- Install and launch Jetpack app
- Navigate to the dashboard
- ✅ Verify that the `Promote with Blaze` card is not shown
- Navigate to Me > App Settings > Debug Setttings
- Enable the `blaze` flag and restart the app
- Navigate back to the dashboard
- ✅ Verify that the `Promote with Blaze` card is shown and it's positioned after the writing prompt and/or before the post cards

## Regression Notes
1. Potential unintended areas of impact
The blaze card is shown when it shouldn't be

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and new unit tests

3. What automated tests I added (or what prevented me from doing so)
`PromoteWithBlazeCardBuilderTest` and `PromoteWithBlazeCardSourceTest`

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by AjeshRPai — APPROVED** [review]
Hey @zwarm , 👋🏼 

Everything looks good to me. 

I have tested the scenarios 🟢 
- [x] flag `blaze` turned off - Verify that Card is not shown 
- [x] flag `blaze` turned on - Verify Card is shown (🌞  Light and 🌑 Dark Mode)

I have also 
- [x] Looked through the code changes. 
- [x] Validatewerehe test cases and checked if they are passing on local  
- [x] Made some changes to fix the lint 

I can see some UI issues, but since you plan to tidy up the UI in another PR, I assume those issues will be addressed in that PR. 

The CI step - Instrumented tests are failing right now. The failure is not because of the changes in this PR. I have retried it again now. I will approve the PR but wait until the CI succeeds before merging.

**Comment by AjeshRPai** [comment]
Hey @zwarm 👋🏼 

After getting a ☕  and Investigating the cause of Instrumented tests failure again, we need to add a mock file for the endpoint as I did in this [commit](https://github.com/wordpress-mobile/WordPress-Android/pull/17101/commits/a75694e62b89536fbfea7747eb815b333e2fcfab).

~I didn't have enough time to do that change. Sorry for that~

I have added the mock mapping file in commit [a2bf946](https://github.com/wordpress-mobile/WordPress-Android/pull/17974/commits/a2bf9466381fed3fa2f51553f43101fad9c9ba37).

If the CI is successful after this change, please merge. 👍🏼

**Comment by zwarm** [comment]
@AjeshRPai 

You beat me to the punch. My plan for today was to separate the PR and see where the test was failing (at which trunk merge). I 🙇 down to your greatness!!! :)

**Pull request merged**
